src/main/java/uk/gov/companieshouse/company/links/processor/InsolvencyStreamProcessor.java

```diff
@@ -65,7 +67,28 @@ public void process(Message<ResourceChangedData> resourceChangedMessage) {
             handleResponse(HttpStatus.valueOf(response.getStatusCode()), logContext,
                     "Response from GET call to company profile api", logMap, logger);
 
-            // TODO DSND-375: check if company needs updating - use response.getData()
+            Links responseLinks = response.getData().getData().getLinks();
+            if (responseLinks.getInsolvency() != null) {
+                logger.trace(String.format("Company profile with company number %s,"
+                        + " already contains insolvency links, will not perform patch",
+                        companyNumber));
+                return;
+            }
+
+            logger.trace(String.format("Current company profile with company number %s,"
+                    + " does not contain an insolvency link, attaching an insolvency link",
+                    companyNumber));
+
+            Data data = response.getData().getData();
+            Links links = data.getLinks();
```

Comment:
You have declared the following: 
```
Links responseLinks = response.getData().getData().getLinks();
Data data = response.getData().getData();
Links links = data.getLinks();
```

We probably can get away with declaring just data and links higher up and removing reponseLinks, bit redundant